### PR TITLE
IA-3223 Stop persisting all instances

### PR DESCRIPTION
**Pull request by Qi77Qi**
https://broadworkbench.atlassian.net/browse/IA-3223

In https://github.com/DataBiosphere/leonardo/pull/2554, we moved away from retrieving all instances along with retrieving a runtime. This PR avoids persisting saving worker instances all tgt.

Tested in fiab, after creating a dataproc cluster
```
mysql> select * from INSTANCE;
+----+-----------+---------------------+---------------+-----------------------------------------------+---------------------+---------+---------------+--------------+----------------------------+
| id | clusterId | googleProject       | zone          | name                                          | googleId            | status  | ip            | dataprocRole | createdDate                |
+----+-----------+---------------------+---------------+-----------------------------------------------+---------------------+---------+---------------+--------------+----------------------------+
|  1 |         7 | terra-test-9bc2bbf6 | us-central1-f | saturn-d2e33cd0-6016-4618-8c01-3e9fa11a240e-m | 9061895619177121550 | Running | 35.239.28.224 | Master       | 2022-02-22 20:23:02.110000 |
+----+-----------+---------------------+---------------+-----------------------------------------------+---------------------+---------+---------------+--------------+----------------------------+
1 row in set (0.00 sec)
```

---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green (you can re-run automation tests with a comment saying `jenkins retest`
- [ ] Run the automation tests multiple times in parallel to weed out instability if applicable via a comment saying `multi-test`
- [ ] Squash and merge; Delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Review by jdcanas — APPROVED** [review]

**Comment by kyuksel** [comment]
Now that we won't be persisting workers and looking at the schema of `INSTANCE` as updated by this PR, I see that `ip` is the only field that isn't currently in `CLUSTER` or `RUNTIME_CONFIG`. Does it make sense to roll `ip` into `RUNTIME_CONFIG` and do away with `INSTANCE` altogether?

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/BaseCloudServiceRuntimeMonitor.scala [hunk 1]
```diff
@@ -103,12 +103,11 @@ abstract class BaseCloudServiceRuntimeMonitor[F[_]] {
   def failedRuntime(monitorContext: MonitorContext,
                     runtimeAndRuntimeConfig: RuntimeAndRuntimeConfig,
                     errorDetails: RuntimeErrorDetails,
-                    instances: Set[DataprocInstance])(
+                    masterInstance: Option[DataprocInstance])(
```
Comment (kyuksel) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/BaseCloudServiceRuntimeMonitor.scala:106]: I meant to mention in previous PRs that the term `master` has implications that can be painful to various demographics. In the UI we started using `main`. I wonder if we could do that in Leo and elsewhere too.

**Comment by Qi77Qi** [comment]
@kyuksel will do renaming in a separate PR just so this will catch tmr's release

> Now that we won't be persisting workers and looking at the schema of INSTANCE as updated by this PR, I see that ip is the only field that isn't currently in CLUSTER or RUNTIME_CONFIG. Does it make sense to roll ip into RUNTIME_CONFIG and do away with INSTANCE altogether?

we do persist hostIp in `CLUSTER`....It might be worth further dropping the table entirely, can you create a ticket for this...

**Comment by kyuksel** [comment]
> It might be worth further dropping the table entirely, can you create a ticket for this

Sure: https://broadworkbench.atlassian.net/browse/IA-3226

**Pull request merged**